### DISCUSSION TITLE
act function in [@types/react-dom] should have a type parameter

### DISCUSSION THREAD

**Discussion by rosczja** [DefinitelyTyped/DefinitelyTyped#60493] [Issues with a @types package]
Currently the act function from react-dom/test-utils is typed as follows:

```
export function act(callback: () => Promise<void>): Promise<undefined>;
```

whereas the flow type in the React repository looks like:

```
export function act<T>(callback: () => T | Thenable<T>): Thenable<T>
```

So I would like to have the same generic parameter `T` in the Typescript definition, so that I can type the return value of the returned Promise or am I missing something?

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
I think the `act` returning the value of the callback is an internal version. Could you check if the `act` from `react-dom/test-utils` actually returns the value from the callback? The documentation could also be updated in that case e.g. https://reactjs.org/docs/testing-recipes.html#multiple-renderers

**Reply by rosczja** [reply]
I am pretty confident, I used this in my code and it seems to work.
